pkg/cache/cluster.go

```diff
@@ -1032,6 +1050,12 @@ func (c *clusterCache) onNodeRemoved(key kube.ResourceKey) {
 			if len(ns) == 0 {
 				delete(c.nsIndex, key.Namespace)
 			}
+
+			nsAll, ok := c.nsIndex[""]
```

Comment:
I don't understand this line. Why `c.nsIndex[""]` and then delete by `key` instead of `""`.

Reply:
Sorry for the confusion. The `nsIndex` is a map of map and I was using a reference of the internal map in `IterateHierarchy()` . Since it was updating the original `nsIndex` I was removing the updated element here. 

I have now changed it to use a copy of the internal map instead of a reference so that we get rid of this update here.